src/main/kotlin/com/itmo/microservices/demo/order/impl/service/OrderServiceImpl.kt

```diff
@@ -191,6 +191,50 @@ class OrderServiceImpl(private val orderRepository: OrderRepository,
         metricsCollector.currentAbandonedOrderNumGauge.set(refundCount)
     }
 
+    @Scheduled(fixedRate = 10000)
+    fun checlOrdersStatus() {
+        val orders = orderRepository.findAll();
+
+        var collectingCount = 0
+        var discardCount = 0
+        var bookedCount = 0
+        var paidCount = 0
+        var shippingCount = 0
+        var refundCount = 0
+        var completedCount = 0
+
+        for (order in orders) {
+            if (order.status == OrderStatus.COLLECTING) {
+                collectingCount++
+            }
+            if (order.status == OrderStatus.DISCARD) {
+                discardCount++
+            }
+            if (order.status == OrderStatus.BOOKED) {
+                bookedCount++
+            }
+            if (order.status == OrderStatus.PAID) {
+                paidCount++
+            }
+            if (order.status == OrderStatus.SHIPPING) {
+                shippingCount++
+            }
+            if (order.status == OrderStatus.REFUND) {
+                refundCount++
+            }
+            if (order.status == OrderStatus.COMPLETED) {
+                completedCount++
+            }
+        }
```

Comment:
This can be implement with `when`.
```suggestion
           when(order.status){
                OrderStatus.COLLECTING -> collectingCount++
                OrderStatus.DISCARD -> discardCount++
                OrderStatus.BOOKED -> bookedCount++
                OrderStatus.PAID -> paidCount++
                OrderStatus.SHIPPING -> shippingCount++
                OrderStatus.REFUND -> refundCount++
                OrderStatus.COMPLETED ->completedCount++
            }
        }
```